crates/bevy_transform/src/systems.rs

```diff
@@ -2,76 +2,96 @@ use crate::components::{GlobalTransform, Transform};
 use bevy_ecs::{
     entity::Entity,
     query::{Changed, With, Without},
-    system::Query,
+    system::{Local, Query},
 };
 use bevy_hierarchy::{Children, Parent};
 
+/// Used for [`transform_propagate_system`]. Ignore otherwise.
+pub(crate) struct Pending {
+    parent: *const GlobalTransform,
```

Comment:
So, I'm not sure this is needed. Confusingly, the `static trait bound doesn't actually imply that all values stored in the type must have a static lifetime.

Instead, it means that the type *itself* must have a static lifetime (unlike e.g. a closure, which can be dropped). https://doc.rust-lang.org/rust-by-example/scope/lifetime/static_lifetime.html#trait-bound

Reply:
Using the following definition, you still cannot find a lifetime that works for the implementation of the system:

```rust
struct Pending<'a> {
  parent: &'a GlobalTransform
  child: Entity,
  changed: bool,
}
```

 1.  The closures used  for `extend` and `for_each_mut` do not allow you to make a global stack that is reused from iteration to iteration.
 2. `'static` cannot be used.
 3. To use as a local, using `Local<Vec<Pending<'a>>>` does not implement `Resource` due to having less than `'static` lifetime.
 
Example error:

 ```
error[E0597]: `global_transform` does not live long enough
  --> crates\bevy_transform\src\systems.rs:67:22
   |
59 |       let mut pending = Vec::with_capacity(*prev_len);
   |           ----------- lifetime `'1` appears in the type of `pending`
...
66 |           pending.extend(children.iter().map(|child| Pending {
   |  ____________________________________________-------_-
   | |                                            |
   | |                                            value captured here
67 | |             parent: &global_transform,
   | |                      ^^^^^^^^^^^^^^^^ borrowed value does not live long enough
68 | |             changed,
69 | |             child: *child,
70 | |         }));
   | |_________- returning this value requires that `global_transform` is borrowed for `'1`
...
90 |       });
   |       - `global_transform` dropped here while still borrowed

error[E0499]: cannot borrow `transform_query` as mutable more than once at a time
  --> crates\bevy_transform\src\systems.rs:75:17
   |
59 |       let mut pending = Vec::with_capacity(*prev_len);
   |           ----------- lifetime `'1` appears in the type of `pending`
...
75 |                   transform_query.get_mut(current.child)
   |                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ `transform_query` was mutably borrowed here in the previous iteration of the loop
...
82 |                       pending.extend(children.iter().map(|child| Pending {
   |  ________________________________________________________________-
83 | |                         parent: &global_transform,
84 | |                         changed: current.changed,
85 | |                         child: *child,
86 | |                     }));
   | |_____________________- returning this value requires that `transform_query` is borrowed for `'1`

error: captured variable cannot escape `FnMut` closure body
  --> crates\bevy_transform\src\systems.rs:82:64
   |
48 |       mut transform_query: Query<
   |       ------------------- variable defined here
...
59 |       let mut pending = Vec::with_capacity(*prev_len);
   |           ----------- inferred to be a `FnMut` closure
...
75 |                   transform_query.get_mut(current.child)
   |                   --------------- variable captured here
...
82 |                       pending.extend(children.iter().map(|child| Pending {
   |  ________________________________________________________________^
83 | |                         parent: &global_transform,
84 | |                         changed: current.changed,
85 | |                         child: *child,
86 | |                     }));
   | |_____________________^ returns a reference to a captured variable which escapes the closure body
   |
   = note: `FnMut` closures only have access to their captured variables while they are executing...
   = note: ...therefore, they cannot allow references to captured variables to escape

error[E0597]: `global_transform` does not live long enough
  --> crates\bevy_transform\src\systems.rs:83:34
   |
59 |       let mut pending = Vec::with_capacity(*prev_len);
   |           ----------- lifetime `'1` appears in the type of `pending`
...
82 |                       pending.extend(children.iter().map(|child| Pending {
   |  ________________________________________________________-------_-
   | |                                                        |
   | |                                                        value captured here
83 | |                         parent: &global_transform,
   | |                                  ^^^^^^^^^^^^^^^^ borrowed value does not live long enough
84 | |                         changed: current.changed,
85 | |                         child: *child,
86 | |                     }));
   | |_____________________- returning this value requires that `global_transform` is borrowed for `'1`
...
90 |       });
   |       - `global_transform` dropped here while still borrowed
   ```